README.md

```diff
@@ -187,21 +192,59 @@ export default Route.extend(InfinityRoute, {
     <h2>{{product.description}}</h2>
   {{/each}}
 
-  {{infinity-loader infinityModel=model.products infinityLoad=(action "loadMoreProduct")}}
+  {{infinity-loader infinityModel=model.products}}
 <section>
 ```
 
-The ability to use closure actions will be available in the `1.0.0-beta` series.  Also, this method uses Controllers.  Despite what you may have heard, Controllers are a great primitive in Ember's ecosystem.  Their singleton nature is great for handling queryParams and actions propagated from somewhere in your component tree.
-
 
 ### Service Methods
 
-The infinity-loader service exposes 4 methods:
+The infinity service also exposes 5 methods to fetch & mutate your collection:
+
+1. model
+2. replace
+3. flush
+4. pushObjects
+5. unshiftObjects
+
+The `model` hook (similar to the Route Mixin `infinityModel` hook) will fetch the first page you request and pass the result to your template.
+
+```js
+import Route from '@ember/routing/route';
+import InfinityRoute from "ember-infinity/mixins/route";
+
+export default Route.extend(InfinityRoute, {
+  model() {
+    return this.infinity.model('product');
+  }
+});
+```
+
+Moreover, if you want to intelligently cache your infinity model, pass `{ infinityCache: timestamp }` and we will return the cached collection if the future timestamp is less than the current time (in ms) if your users revisit the same route.
+
+```js
+import Route from '@ember/routing/route';
+import InfinityRoute from "ember-infinity/mixins/route";
+
+export default Route.extend(InfinityRoute, {
+  model() {
+    return this.infinity.model('product', { infinityCache: 36000 }); // timestamp expiry of 10 minutes (in ms)
+  }
+});
+```
 
-1. replace
-2. flush
-3. pushObjects
-3. unshiftObjects
+Additionally, if you have multiple routes requesting the same model type, in order to intelligently cache your records, you can pass a `label` property.
+
+```js
+import Route from '@ember/routing/route';
+import InfinityRoute from "ember-infinity/mixins/route";
+
+export default Route.extend(InfinityRoute, {
+  model() {
+    return this.infinity.model('product', { infinityCache: 36000, label: 'products-main' }); // label to uniquely identify this collection vs another product collection
```

Comment:
I see you've got `label` defined here, but I also see your method `hashifyInfinityCollection`.

I was originally suggesting we hashify the arguments as a "unique signature" to identify the collection in place of needing a label. Are you set on label being the "right approach" to refer to a collection?

By hashify - I mean reduce the arguments object into a unique alphanumeric string. I hope i've properly explained that as it more or less deprecates the need for an extra label argument.  If adding something to the function signature, I want to be 100% sure it's the best approach.

In addition -- I think the name `hashifyInfinityCollection` is confusing - I think it should be called `cacheInfinityCollection`

Reply:
Lmk what you think about this solution and where I might be missing something - https://github.com/ember-infinity/ember-infinity/pull/287/commits/a04c4930c58d634bdc9236dd21914ddac751fd49#diff-bc59f83659a36857db234e9d3f6a7ee5R268